File: Software Development Module/Week4/PayApp.java
/*
PayApp.java
Billy Cussen 
09/07/2020
*/

//This is the executable class that invokes Pay.java

import java.util.Scanner;
import java.util.InputMismatchException;

public class PayApp {
    
    public static void main(String[]args){

        //Variables
        double payRatePerHour = 0.0;
        double hoursPerWeek = 0;
        double overtimeHoursPerWeek = 0;
        double totalPayPerWeek;
        boolean inputCheck;

        //Objects
        Pay myPay;
        Scanner keyboard;

        //Input
        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter your Pay Rate Per Hour: ");
                payRatePerHour = keyboard.nextDouble();
            } catch(InputMismatchException e){
                System.out.println("Please enter a Number");
                inputCheck = false;
            }
        }while(!inputCheck);

        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter How many Hours you have worked this week: ");
                hoursPerWeek = keyboard.nextDouble();
            } catch(InputMismatchException e){
                System.out.println("Please enter a Number");
                inputCheck = false;
            }
        }while(!inputCheck);

        do{
            inputCheck = true;
            keyboard = new Scanner(System.in);

            try{
                System.out.print("Please enter How many Overtime Hours you have worked this week: ");
                overtimeHoursPerWeek = keyboard.nextDouble();
            } catch(InputMismatchException e){
                System.out.println("Please enter a Number");
                inputCheck = false;
            }
        }while(!inputCheck);

        keyboard = null;
        //Process
        myPay = new Pay(payRatePerHour,hoursPerWeek,overtimeHoursPerWeek);
        myPay.computeTotalPay();

        //Output
        totalPayPerWeek = myPay.getTotalPay();
        myPay = null;
        System.out.println("Your total Pay for the week is: "+totalPayPerWeek);
    }

}

File: Python Course/Software Development Module - in Python/Week4/PayApp.py
"""
PayApp.py
Billy Cussen
27/09/2020
"""

import gc
import Pay
from Pay import myPay

repeat = "y"

while repeat == "y":

    inputCheck = True

    #Input - Hours Worked (Regular)
    while inputCheck:
        try:
            hours = float(input("Please enter your Regular Hours worked this week: "))
        except:
            print("You need to enter a number!")
        else:
            if hours < 0:
                print("You need to enter a number greater than 0")
            else:
                inputCheck = False

    #Input - Hours Worked (Overtime)
    while inputCheck == False:
        try:
            overtime = float(input("Pleas enter your Overtime Hours worked this week: "))
        except:
            print("You need to enter a number!")
        else:
            if overtime < 0:
                print("You need to enter a number greater than 0")
            elif hours < 40 and overtime != 0:
                print("You can't enter Overtime if you've worked less than 40 hours, enter 0 please")
            else:
                inputCheck = True

    #Input - Rate of Pay
    while inputCheck:
        try:
            payRate = float(input("Please enter your Rate of Pay Per Hour: "))
        except:
            print("You need to enter a number!")
        else:
            if payRate < 0:
                print("You need to enter a number greater than 0")
            elif payRate < 10.10:
                print("You can't enter under Minimum Wage in Ireland")  
            else:
                inputCheck = False

    #Process & Output
    myPayApp = myPay(hours,overtime,payRate)
    myPayApp.calculatePay()
    print("Your Regular Pay: "+str(myPayApp.totalRegularPay))
    print("Your Overtime Pay: "+str(myPayApp.totalOverTime))  
    print("TOTAL PAY: "+str(myPayApp.totalPay))

    #Input - Use the App Again?
    repeat = ""
    while repeat != "y" and repeat != "n":
        repeat = str(input("Please enter if you would like to retry or not(\"Y\" or \"N\"): "))
        repeat = repeat.lower()
        if repeat != "y" and repeat != "n":
            print ("You need to enter Y or N!")
        
print ("Thanks for using my App!")
            